client/failover.go

```diff
@@ -0,0 +1,91 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+	"github.com/drand/drand/log"
+)
+
+const defaultFailoverGracePeriod = time.Second * 5
+
+// NewFailoverWatcher creates a client whose Watch function will failover to
+// Get-ing new randomness if it does not receive it after the passed grace period.
+func NewFailoverWatcher(core Client, group *key.Group, gracePeriod time.Duration, l log.Logger) Client {
+	if gracePeriod == 0 {
+		gracePeriod = defaultFailoverGracePeriod
+	}
+
+	return &failoverWatcher{
+		Client:      core,
+		group:       group,
+		gracePeriod: gracePeriod,
+		log:         l,
+	}
+}
+
+type failoverWatcher struct {
+	Client
+	group       *key.Group
+	gracePeriod time.Duration
+	log         log.Logger
+}
+
+// Watch returns new randomness as it becomes available.
+func (c *failoverWatcher) Watch(ctx context.Context) <-chan Result {
+	var latest uint64
+	ch := make(chan Result, 5)
+
+	sendResult := func(r Result) {
+		if latest >= r.Round() {
+			return
```

Comment:
worth a log.debug of skipping duplicates

Reply:
this was added to https://github.com/drand/drand/pull/363